src/managers/NetworkRequestManager.ts

```diff
@@ -168,23 +184,41 @@ class NetworkRequestManager {
 		script: string | undefined,
 		requestId: string,
 		response?: EndpointResponse | undefined,
+		auditInfo?: {
+			log: AuditLog;
+			scriptType: Exclude<RequestEvent['eventType'], 'request'>;
+			associatedId: string;
+		},
 	): Promise<string | undefined> {
 		if (script && script != '') {
 			try {
-				const sprocketPan = getScriptInjectionCode(requestId, response);
+				auditInfo &&
+					auditLogManager.addToAuditLog(auditInfo.log, 'before', auditInfo.scriptType, auditInfo.associatedId);
+				const sprocketPan = getScriptInjectionCode(requestId, response, auditInfo?.log);
 				const _this = globalThis as any;
 				_this.sp = sprocketPan;
 				_this.sprocketPan = sprocketPan;
 				_this.fetch = fetch;
 				const jsScript = ts.transpile(script);
 				const scriptTask = evalAsync(jsScript);
 				await asyncCallWithTimeout(scriptTask, Constants.scriptsTimeoutMS);
+				auditInfo &&
+					auditLogManager.addToAuditLog(auditInfo.log, 'after', auditInfo.scriptType, auditInfo.associatedId);
 			} catch (e) {
 				const errorStr = JSON.stringify(e, Object.getOwnPropertyNames(e));
-				return JSON.stringify({
+				const returnError = JSON.stringify({
 					...JSON.parse(errorStr),
 					errorType: `Invalid ${response != undefined ? 'Post' : 'Pre'}-request Script`,
```

Comment:
In most cases I think it's more reliable/readable to do response != null (or alternatively, response == null and switch the predicates to avoid double-negative thinking).

Reply:
readable sure, reliable nah. Will make the change

Reply:
FYI it does actually return undefined so undefined is more correct here